File: edu/HRO/CMITDW Taal der Wiskunde/o73.java
// opdracht 7.3; puzzel 3
public class o73
{
	public static int list[] = new int[10];
	
	public static void main(String[] args)
	{
		for (int c=2; c<=100; c++)
		{
			clear(list); // clear array
			
			for (int a=2; a<c; a++)
			{
				double b = Math.sqrt(Math.pow(c,2) - Math.pow(a,2));
				if (b%1 == 0 && !check(list, a))
				{
					insert(list, (int) b); // filter duplicate/mirror values
					System.out.println(a+"\t"+(int) b+"\t"+c);
				}
			}
		}
	}
	
	public static void clear(int t[])
	{
		for (int i=0; i<t.length; i++)
			t[i] = 0;
	}
	
	public static void insert(int t[], int a)
	{
		for (int i=0; i<t.length; i++)
		{
			if (t[i] == 0)
			{
				t[i] = a;
				break;
			}
		}
	}
	
	public static boolean check(int t[], int a)
	{
		for (int v : t)
		{
			if (v == a)
				return true;
		}
		return false;
	}
}

File: edu/HRO/CMITDW Taal der Wiskunde/Lua/o73.lua
-- opdracht 7.3; puzzel 3
local i = 0
local t = {} -- associative array

local function wipe(t)
	for k in pairs(t) do
		t[k] = nil
	end
end

for c = 2, 100 do
	wipe(t) -- clear table
	for a = 2, c-1 do
		local b = math.sqrt(c^2 - a^2)
		if b%1 == 0 and not t[a] then
			i = i + 1
			t[b] = true -- filter mirror value
			print(a, b, c)
		end
	end
end
